src/UI/Component/Dropzone/File/File.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace ILIAS\UI\Component\Dropzone\File;
+
+use ILIAS\UI\Component\Component;
+use ILIAS\UI\Component\Droppable;
+
+/**
+ * Interface File
+ *
+ * A dropzone where one can drop files on it to be uploaded on the server.
+ *
+ * @author  nmaerchy <nm@studer-raimann.ch>
+ *
+ * @package ILIAS\UI\Component\Dropzone\File
+ */
+interface File extends Component, Droppable {
+
+	/**
+	 * Get a dropzone like this where the files are uploaded to the given URL.
+	 *
+	 * @param string $url
+	 * @return $this
+	 */
+	public function withUploadUrl($url);
+
+	/**
+	 * Get the upload URL where the files are uploaded.
+	 *
+	 * @return string
+	 */
+	public function getUploadUrl();
+
+	/**
+	 * Get a dropzone like this only accepting the submitted file types for uploading, e.g.
+	 * ['jpg', 'png', 'gif'] to allow some image formats.
+	 *
+	 * @param array $types
+	 * @return $this
+	 */
+	public function withAllowedFileTypes(array $types);
```

Comment:
Should we abstract those and move the abstraction to the src/Data folder (like color and data size)?